src/uct/sm/mm/mm_pd.h

```diff
@@ -18,16 +18,37 @@
 /* Shared memory ID */
 typedef uint64_t uct_mm_id_t;
 
+
+/*
+ * Descriptor of the mapped memory
+ */
+typedef struct uct_mm_mapped_desc {
+    void *address;    /**< local memory address */
+    uint64_t cookie;  /**< cookie for mmap, xpmem, etc. */
+} uct_mm_mapped_desc_t;
+
+
 /*
  * Memory mapper operations - MM uses them to implement PD and TL functionality.
  */
 typedef struct uct_mm_mapper_ops {
+
+    ucs_status_t (*query)();
+
+    ucs_status_t (*reg)(void *address, size_t size, 
+                        uct_mm_id_t *mmid_p);
+
+    ucs_status_t (*dereg)(uct_mm_id_t mm_id);
+
     ucs_status_t (*alloc)(size_t *length_p, ucs_ternary_value_t hugetlb,
                           void **address_p, uct_mm_id_t *mmid_p UCS_MEMTRACK_ARG);
 
-    ucs_status_t (*attach)(uct_mm_id_t mmid, void **address_p);
+    ucs_status_t (*attach)(uct_mm_id_t mmid, size_t length, 
+                           void *rem_addr, uct_mm_mapped_desc_t **mm_desc);
```

Comment:
since mmap will also add info, i think the mm_desc should not be allocated by the specific transport, but rather on the mm level. the transport will fill it.